Automattic/simplenote-macos
# Interlinks Autocomplete UI: Mark I

### Description:
This PR implements the Interlink Autocomplete UI, and takes care of all of the logic required to display the Autocomplete Window below the Editor's cursor, when appropriate.

**Note:** Actual Interlink insertion will come up in a follow up PR!

@aerych Would you be game for this one?
Thanks in advance!!

P.s.: Apologies about the size of the PR, new Storyboard included!!

Ref. #663

### Details:
For the record! We're using a NSWindow instance for Autocomplete purposes, because the Autocomplete Feature must:

A. Not interfere with Text Input in the main editor. You must be able to go on typing
B. Not be affected by the ScrollView
C. And it must also not be affected, anyhow, by anything else in the hierarchy


### Test: Interface!
1. Open any note
2. Enable Markdown
3. Type `Hello 🇮🇳 🌎 [a`  (any emoji would work!)

- [x] Verify a Window shows up below the `a` character, with 3 placeholder rows
- [x] Verify the Autocomplete Window has rounded corners (macOS +10.15)
- [x] Verify that the mouse cursor switches to *Pointing Hand*, when hovering over the Interlink Suggestion Window
- [x] Verify that Light / Dark Mode switches affect the autocomplete window

### Test: Window Vertical Positioning
1. Place the Simplenote Window at the bottom of the screen
2. Insert `[a` in the very last line of the editor

- [x] Verify that the autocomplete window shows up above the cursor, and not below (to avoid clipping!)

### Test: Window Horizontal Positioning
1. Place the Simplenote Window close to the right edge of the screen
2. Insert `[a` on the righter-most corner of the editor

- [x] Verify the Autocomplete Window is positioned in such a way that it doesn't get clipped

### Release
These changes do not require release notes.

### Screenshots
<img width="400" alt="Autocomplete" src="https://user-images.githubusercontent.com/1195260/94822941-a242b780-03d9-11eb-97cc-cd0fdad5a11a.png">
